Skeletal Inventory 1st Pull Request

kdaveed opened the pull request:
Now the  menu page _Skeleletal Inventories_ allows to add skeletal inventories for fresh and dry bones. It offers a list of existing skeletal inventories, which are links to their profile pages. Fresh and dry bones have different profile pages, because in the case of dry bones it is possible to add only single bones. Please check if the implementation is satisfactory!

Commit:
TestData is replaced to the CEF directory.

Commit:
pageConfig.js were renamed to tabs.js

Commit:
Slight modification

Commit:
Stashed files back

Commit:
Merged classes and a new image for list elements.

Commit:
Now the systemic part handler gets only the parent,
and from its descriptor it has to set itself.
It checks the existingToSelect field properly and displays
a list of it if we can select one.

Commit:
The CoherentSkeletalRegion is now in a distinct file.
More overseeable.

Commit:
New Checkbox container element.

Commit:
The skeletalSubdivision custom entry form is rewritten

Commit:
generateArray function were merged

Commit:
Triple reference were renamed

Commit:
New checkbox html lib element

Commit:
GenerateArray is not any more in config

Commit:
margin classes for the checkbox

Commit:
Configuration of the skeletalSubdivision page, and calling
from the ftl.

Commit:
New queries and pages

Commit:
The uri is dynamic and the label is loaded through a query

Commit:
The new systemic part adder

Commit:
Extended configuration of the skeletal inventory

Commit:
Final page assembler routines

Commit:
Simplified page loader. There is now switch. The arrived
value in the pageUri parameter is just the name of the
file.

Commit:
Not need any more

Commit:
File rename and some unnecessary files were removed.

Commit:
PageDataPerform is the object oriented version of the
execution.js. It were extended to be able iterate on
keys of data, if the configuration object contains
the "keys" keyword. As well the DataInit discovers
all pageData field recursively (except pageElements) if
there is a field whose value has to be set by an operation.

Commit:
simplified type getter function

Commit:
Now the variables got the this prefix

Commit:
Now if the dataToSelect variable is undefined then the algorithm goes on.

Commit:
More compact data operation definition. The ftl file calls
the queryLib's queries, and the PageDataPerfor with the help
of DataInit interprets this definition and performs
the appropriate queries.

Commit:
I apply $ character to denote that I want to perform
operation on the same field multiple time. PageDataPerform can handle this
new feature. Otherwise each data operation is working properly with this
more compact definition as well.

Commit:
In the page definition there are the mappings to the symmetric classes.

Commit:
The SystemicPartAdder is updated so the class class hierarchy
is changed. Now it works on the singles bone page.

Commit:
The configuration for the cef of single skull bones is
complete.

Commit:
Single Query now can add simple literals to any object.
To the queryLib some new useful queries were added.
DataLoader.java can interpret the subClass2 query.

Commit:
UI for vertebra bone organs is working

Commit:
Two new functions used by the PageDataPerform and
DataInit functions.

Commit:
The configuration of tabs is extended with
the single bone key.

Commit:
Style for the SingleBoneAdder

Commit:
New triple config that helps to particular
single bones on skeletalInventory profile
page.

Commit:
New queries

Commit:
Now the single bone CEF for uppper limb works
properly.

Commit:
Small styling

Commit:
Now the single bone configuations on the profile page is correct for
each skeletal divisions.

Commit:
New lib ftl files

Commit:
FMA prefix

Commit:
Necessary config of skeletal region tabs

Commit:
Querylib update for new types of queries

Commit:
DataInit now can handle list of operations

Commit:
Updated script import

Commit:
A bug fixed.

Commit:
DataTable data getter routine fixed

Commit:
DataField for single bones is set properly

Commit:
Execution is replaced by the PageDataPerform.js

Commit:
Now the skeletal division custom entry form display the proper
entry form by the Skull and the Vertebral Columns. SkeletalRegion.js
and SystemicPartAdder.js were modified accordingly.

Commit:
Skull and Vertebral Column config changes on the skeletal inventory
profile page so that they point to the appropriate custom entry form.

Commit:
Deleted Files

Commit:
This is the data configuration of the skeletal division profile page.

Commit:
New query for the vertebral column.

Commit:
sortToKey function is now implemented.

Commit:
Delete files

Commit:
Bony Pelvis is configured as skeletal division

Commit:
PageDataPerform now can handle switch case in the query variable.

Commit:
The switch case is applied on the query name

Commit:
New prefix for the switch case

Commit:
SkeletalRegion now redirects correctly after the bone saving process
to the skeletalDivisionPage. SkeletalDivision page is now
configured correctly.

Commit:
The query for the skeletal division is set

Commit:
This file is necessary for the single bone custom entry forms.

Commit:
SingleBoneCEF controller is there for the
single bone custom entry form. The querLib is used by the
dataOperationConfig is for query definition needed for the CEF.

Commit:
skeletalDivision was replaced by skeletalSubdivision. Moreover
two ftl modifications.

Commit:
Now the skeletalRegions are configured and displayed correctly on
the skeletal inventory page.

Commit:
Configuration modifications so that that symmetric skeletal
regions call the symmetricCoherentSkeletalRegion CEF
with the parameter skeletalRegion.

Commit:
Simplified data configuration for symmetric and
partly symmetric data operations.

Commit:
The necessary queryLib extensions

Commit:
Config modification

Commit:
Two ftl for the two custom entry form

Commit:
Now the coherent skeletal regions appear properly on
the skeletal inventory profile page by the partly
symmetric bones.

Commit:
PageInit was updated previously. It just now commited

Commit:
New initial array for the custom page data opeations.

Commit:
SkeletalRegion.js were extended to handle the
the CoherentSkeletalRegions. It is called by Skull,
Vertebral Column, Bony Pelvis.

Commit:
The skeletal division configurations.

Commit:
In the case of Vertebral Column we have to restrict the
between the systemic parts there is no BoneOrgan
subClass is enabled.

Commit:
Deleted Files

Commit:
For the symmetric bones there is as well a class.
Can be called properly from the skeletal inventory page.
EntryFormTypes is deleted. Config files are updated.

Commit:
Now adding the symmetric skeletal region is now working.

Commit:
Global commit for config file modifications.

Commit:
Back-end change

Commit:
Small styling

Commit:
Module Editors

Commit:
Container is insensitive if the localData is not defined.

Commit:
There is only one tab for the different skeletal regions.
Under the tabs we can add skeletalRegions, coherentSkeletalRegions
and single bones

Commit:
Adding existing bone organs to skeletal division is now
possible.

Commit:
Even if the the this.dataSet.existing is not defined the
program goes on.

Commit:
Single Bones of Lower Limb is properly configured

Commit:
Bug fixed

Commit:
Important modification for the case if one of the
subvariable in the variable chain is undefined.

Commit:
So the following type configuration
can be applied

Commit:
Now the addition of the existing bone organs is working.

Commit:
Log remove and code styling

Commit:
Styling

Commit:
New msg function for the popupcontroller

Commit:
Images

Commit:
Bugs fixed, and now the addition of existing bone organs is
working propertly. Moreover if all possible bone organ were added, then the
list icon disappears, and if the last existing we added
then the list disappers as well.

Commit:
Removed log

Commit:
PageDataPerform extensions. Now if the query has the
parameter sorted, then it creates automatically an object.

Commit:
Small page config modifications.

Commit:
The Bone Organ field now handles properly the existing
bone organs.

Commit:
SystemicPartAdder now considers properly if existing
coherent skeletal division we added.

Commit:
The query definition were restructured and now it is clearer.

Commit:
Now the addition of bone organs to existing  skeletal divisions
is now working.

Commit:
Java updates

Commit:
Bug fixed, no to single bone we can add existing properly.

Commit:
Now the SystemicPartAdder handles the the existing bone
organs proporly even by not singleSelect case.

Commit:
Now the PageDataPerform goes on even if an input array
has the length zero

Commit:
Duplicate file

Commit:
Unnecessary files

Commit:
Unnecessary file

Commit:
Deleted file

Commit:
Config files of skeletal subdivisions are moved to
skeletalInventory/coherentSkeletalDivision folder.

Commit:
symmetricSkeletalDivision are placed to the
skeletalInventory/symmetricSkeletalDivision folder.

Commit:
Now we can merge arrays so that we select a list
of key from the array object.

Commit:
CoherentSkeletalRegion is renamed to
CoherentSkeletalSubdivision.

Commit:
SkeletalRegion was renamed to CoherentSkeletalDivision

Commit:
partlySymmetricCoherentConfig were placed into the
CEF/skeletalSubdivision folder.

Commit:
Deleted files

Commit:
Deleted files

Commit:
partlySymmetricConfig.js is renamed to config and
placed into the coherentSkeletalSubDivision folder.

Commit:
symmetriCoherentCEF.js is renamed to config.js and placed
into the symmetricSkeletalDivision folder.

Commit:
Deleted files

Commit:
Added queries

Commit:
boneDivision folder was renamed to skeletalDivision

Commit:
query file was simplified and renamed to pageData.js

Commit:
Deleted files

Commit:
There is only one container with the tabs of
different skeletal regions.

Commit:
Added files

Commit:
Now from the skeletalInventory profile page the
edit button of the coherentSkeletalDivision points
to the coherentSkeletalDivision profile page.

Commit:
partlySymmetricCoherentConfig.js is placed to the
lib directory.

Commit:
In the CEF now the renamed JavaScript libraries are
called properly and the CEF is can be called from
the skeletalInventory profile page.

Commit:
SkeletalDivisionSelector now works properly and
after the save process the user lands on the
skeletalDivision page.

Commit:
The config files for the two types of skeletal divisions
are set properly.

Commit:
The proper custom entry form type are set

Commit:
skeletalDivision page is configuration is updated

Commit:
Deleted files

Commit:
Deleted files 2

Commit:
Now the link from the skeletal inventory page to
the skeletalDivision page works properly.

Commit:
PageDataPerform handles if a query fails, and it tries
once again. And if it fails again then an error msg
is sent.

Commit:
This file is the new version of the SkeletalRegion. It
was just renamed.

Commit:
SkeletalCEF uri is set to the bone regions.

Commit:
Skeletal Division now displays the data properly
if there is an existing

Commit:
Bug fix and small rename

Commit:
The configuration is updated, so we can now add call
the skeletalDivision CEF from the skeletalDivision profile page.
SkeletalDivisionSelector now passes the correct data to
the SkeletalDivisionSelector.

Commit:
Global commit for the FTL files. They are including the rearranged set
of configuration data properly.

Commit:
DataLoader has been made to more clear.

Commit:
The dataField are now sending the customEntryForm uris to the
skeletalDivision profile pages.

Commit:
Added query to the queryLib

Commit:
Filter was missing by the skeletelDivision query

Commit:
Page Configuration for coherentSkeletalDivision profile page

Commit:
lowerLimb, shoulder Girdle and upperLimb are now so
configured that after the deletion of the bone organ
from the skelelalDivision, the deleted item
appears on the single list

Commit:
No the editButtonLink controller adds the skeletalDivisionType
to single bones after the deletion so that they can
be identified by the skeletalInventory profile page.

Commit:
Bug fix

Commit:
Bug fix

Commit:
New query

Commit:
Data Input now can label the single bones after
they are deleted from a skeletal division.

Commit:
DataFieldDefinition corrections

Commit:
The CoherentSkeletalSubdivision handles the addition
of existing skeletal divisions properly.

Commit:
Styling

Commit:
Two delete config class were added. They are considered
by the DeleteController. Now it is possible to
delete whole skeletal divisions and coherent skeletal
divisions.

Commit:
Data configuration of the phalanx

Commit:
Adding the phalanx bones of foot is working.
The data the interface work witt is hard coded yet.

Commit:
Some logs and unnecessary parts were removed.

Commit:
Updated upper- and lowerLimb configuration.

Commit:
Config files for skeletal division

Commit:
Config files for coherent skeletal division

Commit:
Now it is possible to add coherent rib cage.

Commit:
Now both the single and coherent rib cage saving
and display works.

Commit:
Bug fixes in the config files so that now
the single bones can be added correctly.

Commit:
Now the completeness button works again.

Commit:
The PhalanxEdit class not considers the cardinalities
properly

Commit:
Deletion of phalanx is considered properly

Commit:
The CustomFormGenerator class is invoked by the form generation
of for bony parts. The universal.ftl gets the form data in
JSON format and the FormGenerator.js generates the form based
on the configuration data. The Selector.js is JS object for
the selector elements and their routines.

Commit:
The RestrictionTriples class is the frame for generating
data sturctures (SPARQL queries) for the forms.
The BonyDataSet1.java contains the definition of the
query required in case of some skeletal regions such as Skull.

Commit:
IndividualTemplateLocator is on the third tier.
The InstanceCreator uses the the IndividualTemplateLocator
to locate the template. The NewInstance is the Servlet that
uses the InstanceCreator.

Commit:
It makes no sense to create form function. Anyway the second
will be called, even without parameter. Moreover new
form related functions were added. The SelectorAddField
is responsible for creating new instances of the selected
class. Teh SelectorField is the helper method for the selector
element of the field. The pageConfig.js is updated to the
correct classes of the skeletal inventories.

Commit:
The following classes serving the configuration and
generation of different dataset appear on entry forms
for dry bones.

Commit:
The classes of the commit are responsible for the modeling
of the entry form elements.

Commit:
SelectorAddField's bug fixed. Now it is configured to
send the proper url pattern.
bonyPartPageConfig.n3 sets the custom template for dry bones.

Commit:
The template for dry bone and the configuration JS files.

Commit:
Just for test label is displayed on the dry bone template.
---
The NewInstance.java class creates the
instance with label generated from the class label and the individual
uri. Moreover it searches for the custom template file saved in
RDF triples.
---
The InstanceCreator.java is not needed any more.

Commit:
pageConfig.js contains the proper configuration for
the menu page. Parameter of the data query is the PrimarySkeletalInventory
class. Moreover the typeLabel field is displayed.
---
DataLoader.java is extended to the instance retriever query.
---
*.js and *.css is styling for the LinkRowTable

Commit:
DefaultPageLoad.java loads the page of any individual based on its
type. It queries the type and after the custom template file.
---
Its mapping appears in the web.xml
---
The skeletalInventoryMenuPage is configured to call this servlet.

Commit:
TabContainer can now be used as container, so local data can be
defined to it.

Commit:
Bugs fixed in the descriptor of the bonyPart page. Now the tabs and
their content appear correctly.

Commit:
The bonyPartSkeletalInventory.ftl contains all the configuration files
for the page.

Commit:
--- data.js
The key "class" as http request parameter has been set
--- tabs.js
formUri-s are set. Uri bug fixed
--- upperLimb.js, lowerLimbs.js
Labels, uri, formUri are set correctly
--- customMenuPage.ftl
seemingly nothing has changed

Commit:
---FormDefinition.java
The super class of each form definitor class
--- FormElementType.java
ENUM for the types
--- RequestData, ConstantRequestData, SubmissionType.java
The implementation of the types that call different enums

Commit:
---Form.java
contains the lists of the submission and redirection definition
as fields
---BonyForm, BonyForm1.java
call the new fields as definition
---universal.ftl
converts the data definition into JSON

Commit:
FormGenerator considers the new definitions.

Commit:
--- BonyForm.java
Parameter bug fixed
--- FormGenerator.js
Submission creator bug fixes, this.element introduced
--- *.ftl
dynamicFormLib.ftl simplifies the Java JSON conversion in
the universal.ftl

Commit:
Form configuration updates

Commit:
Completeness Dataset configuration contains now the correct
uris

Commit:
Getter for dataType has been set

Commit:
--- DataLoader.java
new queries for getting the completeness and completenessStates.
---pageConfig.js
Defines that these queries loads the single data into
the appropriate variables.

Commit:
Tiny FTL corrections

Commit:
Now the Default page load can retrive the superclasses
in the right order with the getSuperClass(..,..) routine
and the while loop.

Commit:
Bone Organ Profile Page is set to have link to its
skeletal Inventory.

Commit:
CoherentSkeletalDivision profile page is set to
have link to its skeletal inventory.

Commit:
SkeletalDivision is set to have link to its skeletalInventory
profile page.

Commit:
DataLoader.java contains the queries for the different
skeletal levels to get the skeletalInventory uri

Commit:
JS bug fixes

Commit:
The skeletalInventory is not any more sent to the page
after the form submission. It will be retrieved by the page
load process.

Commit:
The custom template files for the different types are
refreshed.

Commit:
The processing of the submission of the BonyPart forms are different a
little thus the Triple1 and ConstantNewInstance1 classes were created
to tackle the problem.

Commit:
The configuration classes were updated so the bony part
and the fresh bone forms are working now.

Commit:
In each classes the dataOperation field is updated.

Commit:
The configuration for the Bony Part config and its call from
the DataInput.java

Commit:
Now on the bony part skeletalInventory has the label editor on its
field.

Commit:
The fresh bone skeletalInventory has as well the label editor field.

Commit:
The Delete configurations naming conventions are now clear.
The DeleteController calls properly the new config classes.

Commit:
New data table element, the BinButtonLink has been added.
It is represented by a rubbish bin image that denotes
in our case the complete delete of bone organs or any
skeletal division. The dataTable is prepared so it does
not create any addition table title for it.

Commit:
Config updates

Commit:
BinButton prefix

Commit:
Naming conventions are cleared as well in the dataLoader

Commit:
DefaultPageLoad.java now determines the custom template
for any instance right now.

Commit:
The URI has been updated.

Commit:
The parameters in the skeletal inventory retriever query
has been changed.

Commit:
Java configuration for template generation.

Commit:
JavaScript extensions and bug fixes

Commit:
The bone organ has as well a unique profile page.

Commit:
---DataTable.css
Styling of the font of the data entries.
---*js
Naming corrections of the data table are corrected.

Commit:
New styling for containers and data tabels.
Typo fix for pelvicGridle.js

zarquon42b commented:
Comments:
-  no way to delete skeletal inventories (SI)
- only SI (not dry or fresh bone) working
- new install contains fresh_bone inventory lacking a template file
- missing support of non-primary SIs
- dry bone selection generates a blank page
- fresh bone throws an exception:  

```
Stack trace (full trace available in the vivo log): java.lang.IndexOutOfBoundsException: Index: 0, Size: 0 at java.util.ArrayList.rangeCheck(ArrayList.java:653) at java.util.ArrayList.get(ArrayList.java:429) at edu.cornell.mannlib.vitro.webapp.search.controller.NewInstance.processRequest(NewInstance.java:59) at edu.cornell.mannlib.vitro.webapp.controller.freemarker.FreemarkerHttpServlet.doGet(FreemarkerHttpServlet.java:111) at javax.servlet.http.HttpServlet.service(HttpServlet.java:624) at javax.servlet.http.HttpServlet.service(HttpServlet.java:731) at edu.cornell.mannlib.vitro.webapp.controller.VitroHttpServlet.service(VitroHttpServlet.java:71) at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:303) at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208) at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52) at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241) at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208) at edu.cornell.mannlib.vitro.webapp.filters.PageRoutingFilter.doNonDisplayPage(PageRoutingFilter.java:108) at edu.cornell.mannlib.vitro.webapp.filters.PageRoutingFilter.doFilter(PageRoutingFilter.java:99) at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241) at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208) at edu.cornell.mannlib.vitro.webapp.filters.ClickjackFilter.doFilter(ClickjackFilter.java:31) at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241) at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208) at edu.cornell.mannlib.vitro.webapp.filters.RequestModelsPrep.doFilter(RequestModelsPrep.java:56) at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241) at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208) at edu.cornell.mannlib.vitro.webapp.filters.URLRewriterFilter.doFilter(URLRewriterFilter.java:38) at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241) at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208) at edu.cornell.mannlib.vitro.webapp.filters.JSessionStripFilter.doFilter(JSessionStripFilter.java:60) at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241) at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208) at edu.cornell.mannlib.vitro.webapp.filters.CachingResponseFilter.produceBasicResponse(CachingResponseFilter.java:321) at edu.cornell.mannlib.vitro.webapp.filters.CachingResponseFilter.doFilter(CachingResponseFilter.java:115) at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241) at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208) at edu.cornell.mannlib.vitro.webapp.i18n.selection.LocaleSelectionFilter.doFilter(LocaleSelectionFilter.java:62) at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241) at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208) at edu.cornell.mannlib.vitro.webapp.filters.CharsetEncodingFilter.doFilter(CharsetEncodingFilter.java:34) at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241) at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208) at edu.cornell.mannlib.vitro.webapp.filters.SessionTimeoutLimitingFilter.doFilter(SessionTimeoutLimitingFilter.java:39) at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241) at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208) at edu.cornell.mannlib.vitro.webapp.filters.StartupStatusDisplayFilter.doFilter(StartupStatusDisplayFilter.java:64) at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241) at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208) at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:221) at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:122) at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:505) at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:169) at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:103) at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:956) at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:116) at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:436) at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1078) at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:625) at org.apache.tomcat.util.net.JIoEndpoint$SocketProcessor.run(JIoEndpoint.java:318) at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) at java.lang.Thread.run(Thread.java:745) 
```

Commit:
---LiteralField.js
There is new type of field called LiteralLinkField.
---SWMAP.js and PageElementMap.js
Updated mapping

Commit:
DataLibrary.js now considers the local dataset by the
url generation.

Commit:
The hasPart and isAbout predicates were transposed.

Commit:
New class for deleting the instance with with deleting all
from its type.

Commit:
The NewInstance.java does not search for the template of the individual
it rederects to the defaultPageLoad link where the DefaultPageLoad.java
cares about the loading.

Commit:
If the DeleteController gets as parameter the instance as well then it
will be deleted. Moreover the deleteSkeletalInventory feature is
added.

Commit:
DeleteSkeletalInventory config

Commit:
During the delete process a gif appear to show now currently is
waiting.

Commit:
The type of the table on the skeletal inventory menu page
is now DataTable. The link to its profile page is the
label field, whose type is the LiteralLinkField.
The css is set as well.

Commit:
The TemplateGetter is an individual class that gets
based on the individual uri determines the custom
template. Called by the DefaultPageLoader.java.

Commit:
DeleteConf import deleted

zarquon42b commented on productMods/js/UIModules/pageElements/SelectorAddField.js:
```diff
@@ -0,0 +1,37 @@
+
+var SelectorAddField = function(configData){
+	
+	this.container = html.form("/vivo/newInstance", configData.id)
```
Please make "/vivo/newInstance" dynamic (i.e. if the root directory is not vivo this script does not work).

zarquon42b commented on productMods/js/UIModules/pageElements/SelectorAddField.js:
```diff
@@ -0,0 +1,37 @@
+
+var SelectorAddField = function(configData){
+	
+	this.container = html.form("/vivo/newInstance", configData.id)
```
please make the path universal: if the webapp.name is not vivo, this script does not work.

zarquon42b replied:
Nicely fixed

Commit:
Two files merged
from master.

Commit:
Merging master into skeletalInventory

Commit:
The  Popup window for waiting looks better.
Moreover the hard coded /vivo url was replaced by a variable.

kdaveed commented:
Issues were fixed and merge conflicts were resolved. Branch is ready to pull.

zarquon42b commented:
I am still having issues when creating dry bone/fresh bone SI:
fresh bone => " For this type of instance there is not custom template file defined. "
dry bone => empty page

Pls fix.

zarquon42b commented:
Pls fix #75 ,too

Commit:
lowerlimb.js was renamed to lowerLimb.js

Commit:
The template evalutaion based on individualUri happens through the TemplateGetter.

Commit:
Log files for debug

Commit:
Now the superclass tree is evaluted properly.

Commit:
Updated configuration files for skeletal inventory profile pages.

Commit:
log removed

Commit:
JS and FTL clearing.

kdaveed commented:
@zarquon42b Finally the features are working on the virtual machine. Both the skeletal inventories for dry and fresh bone are working, and as well as the image upload and selection.

zarquon42b commented:
Nice, everything seems to work now. Can you pls remove the plain skeletal inventories (as they are not needed as far as I understand @cuboideum), after @cuboideum confirms the removal request=

cuboideum commented:
Yes please, remove the 'skeletal inventory' option, to the effect that users can just choose between 'fresh bone skeletal inventory' and 'dry bone skeletal inventory'.

Commit:
Primary Skeletal Inventory select option has been removed

kdaveed commented:
@zarquon42b @cuboideum 

Fixed in https://github.com/RDFBones/RDFBones/pull/68/commits/cd9a20af170c766264e1a530ccb041ce89a2cd26

zarquon42b commented:
Finally merged

zarquon42b commented:
@kdaveed: please rebase your other working branches derived from master
